ultradns/resource_ultradns_dirpool.go

```diff
@@ -416,12 +429,36 @@ func populateResourceFromDirpool(d *schema.ResourceData, r *udnssdk.RRSet) error
 	} else {
 		d.Set("conflict_resolve", p.ConflictResolve)
 	}
-
+	log.Infof("r.RData= %v and p.RDataInfo= %v", r.RData, p.RDataInfo)
 	rd := makeSetFromDirpoolRdata(r.RData, p.RDataInfo)
 	err = d.Set("rdata", rd)
 	if err != nil {
 		return fmt.Errorf("rdata set failed: %v, from %#v", err, rd)
 	}
+
+	// Block created to set no_response
+
+	if r.Profile["noResponse"] != nil {
+		//noResponse := r.Profile["noResponse"].(map[string]interface{})
+		var noResponseBlock map[string]interface{}
+		np, err := makeDirpoolRdataInfoAPI(r.Profile["noResponse"])
+		if err != nil {
+			return fmt.Errorf("noResponseBlock unmarshalling failed: %v, from %#v", err, np)
```

Comment:
The unmarshalling will fail in case we got some extra fields coming in from API when there is a DTO upgrade and it will give the error and yup engineer will be able to take corrective actions cause i was also able to debug using the same.